pmd-java/etc/grammar/Java.jjt

```diff
@@ -1712,7 +1719,9 @@ void EqualityExpression() #void:
 void Pattern() #void:
 {}
 {
-    PrimaryPattern() [ GuardedPatternCondition() #GuardedPattern(2) ]
+    LOOKAHEAD(ReferenceType() "(") RecordPattern()
+    | LOOKAHEAD("(") ParenthesizedPattern()
+    | TypePattern() [ LOOKAHEAD({getToken(1).kind == SC_AND && jdkVersion < 19}) GuardedPatternCondition() #GuardedPattern(2) ]
```

Comment:
This lookahead including the GuardedPatternCondition can hopefully be removed with one of the later java versions. Right now, it's only needed for java 18 preview - which parses the source really different...